File: project/app-domain/src/main/java/com/numberbox/appdomain/annotation/UseCase.java
package com.numberbox.appdomain.annotation;

import java.lang.annotation.*;

/**
 * Def. 인스턴스화를 진행함
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface UseCase {
}

File: project/app-domain/src/main/kotlin/com/numberbox/app/domain/system_construction/UseCase.kt
package com.numberbox.app.domain.system_construction

import java.lang.annotation.*
import java.lang.annotation.Retention
import java.lang.annotation.Target

/**
 * Def. 의존주입을 가능하게 함
 * Desc. 스프링의 @ComponentScan 역할을 함
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
annotation class UseCase